corpus.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var request = require('request');
+var lunr = require('lunr');
+
+function Corpora(name, baseurl, indexPath, remote) {
+	
+	this.name = name;
+	this.baseurl = baseurl;
+	this.indexPath = indexPath;
+	this.remote = remote;
+	this.key = name.toLowerCase().split(' ').join('-');
+	this.lastUpdate = null;
+}
+
+Corpora.prototype.load = function() {
+
+	var that = this;
+	request(this.baseurl + this.indexPath, function (error, response, body) {
```

Comment:
[`request-promise`](https://www.npmjs.com/package/request-promise) may be useful here.